javascript/javascript2/week2/movies.js

```diff
@@ -0,0 +1,2628 @@
+        rating: 8.1,
+        votes: 50250,
+        running_times: 9600,
+    },
+    {
+        title: "Anchorman 2: The Legend Continues",
+        year: 2013,
+        rating: 6.3,
+        votes: 143995,
+        running_times: 8580,
+    },
+    {
+        title: "Anchorman: The Legend of Ron Burgundy",
+        year: 2004,
+        rating: 7.2,
+        votes: 288977,
+        running_times: 6240,
+    },
+    {
+        title: "And Now for Something Completely Different",
+        year: 1971,
+        rating: 7.6,
+        votes: 25816,
+        running_times: 5280,
+    },
+    {
+        title: "And So It Goes",
+        year: 2014,
+        rating: 5.8,
+        votes: 10471,
+        running_times: 5640,
+    },
+    {
+        title: "And Soon the Darkness",
+        year: 2010,
+        rating: 5.2,
+        votes: 14637,
+        running_times: 5460,
+    },
+    {
+        title: "And Then There Were None",
+        year: 1945,
+        rating: 7.6,
+        votes: 10271,
+        running_times: 5820,
+    },
+    {
+        title: "Angel Eyes",
+        year: 2001,
+        rating: 5.6,
+        votes: 20391,
+        running_times: 6120,
+    },
+    {
+        title: "Angel Heart",
+        year: 1987,
+        rating: 7.3,
+        votes: 68210,
+        running_times: 6780,
+    },
+    {
+        title: "Angela's Ashes",
+        year: 1999,
+        rating: 7.3,
+        votes: 18063,
+        running_times: 8700,
+    },
+    {
+        title: "Angels & Demons",
+        year: 2009,
+        rating: 6.7,
+        votes: 235481,
+        running_times: 8760,
+    },
+    {
+        title: "Angels in the Outfield",
+        year: 1994,
+        rating: 6.1,
+        votes: 24230,
+        running_times: 6120,
+    },
+    {
+        title: "Angels with Dirty Faces",
+        year: 1938,
+        rating: 8,
+        votes: 17119,
+        running_times: 5820,
+    },
+];
+
+//Create an array of movies containing the movies with a short title
+
+const shortTitle = movies.filter(movie => movie.title.length < 4);
+console.log("the short movies", shortTitle);
+
+//Create an array of movie titles with long movie titles
+const longTitle = movies.filter((movie) => {
+    return movie.title.length > 4
+}).map((movie) => {
+    return movie.title;
+});
+
+console.log("longTitle ", longTitle);
+
+//movies between 1980 and 1989
+
+const moviesBetween1980And1989 = movies.filter((movie => {
+    return movie.year >= 1980 && movie.year <= 1989
+})).length;
+console.log(" count of 1980 and 1989 ", moviesBetween1980And1989);
+
+/** array that has an extra key called tag.
+ *  The tag is based on the rating: 
+ * Good (>= 7), Average (>= 4 and < 7), Bad (< 4) */
+const moviesWitTag = movies.map(movie => {
+    if (movie.rating >= 7) {
+        movie.tag = "Good";
+    }
+
+    else if (movie.rating >= 4 && movie.rating < 7) {
+        movie.tag = "Average";
+    }
+    else if (movie.rating < 4) {
+        movie.tag = "Bad"
+    }
```

Comment:
Use switch-case instead of multiple else-if conditions. Then your code will look more clean and pretty